src/librustc/ty/context.rs

```diff
@@ -1554,7 +1554,7 @@ impl<'gcx: 'tcx, 'tcx> GlobalCtxt<'gcx> {
     /// Call the closure with a local `TyCtxt` using the given arena.
     pub fn enter_local<F, R>(
         &self,
-        arena: &'tcx DroplessArena,
+        arena: &'tcx SyncDroplessArena,
```

Comment:
Could the local arenas in theory be non-synchronized?

Reply:
All the arenas could be thread-local if we find a way to deal with lifting of types.